business/badgeCalculator.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2018, The Linux Foundation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+// Responsible for calculating what badge a component/definition has.
+// For now we are just going to go with the simple:
+// If license/copyright there it gets a 3. Only one of those it gets a 2. None, it gets a 1
+
+const _ = require('lodash');
+
+const scoreToUrl = {
+  1: 'https://img.shields.io/badge/Clearly%20Defined%20Score-1-red.svg',
+  2: 'https://img.shields.io/badge/Clearly%20Defined%20Score-2-yellow.svg',
+  3: 'https://img.shields.io/badge/Clearly%20Defined%20Score-3-brightgreen.svg'
+};
+class BadgeCalculator {
+  constructor(definition) {
+    this.definition = definition;
+  }
+
+  getBadgeUrl() {
+    return scoreToUrl[this.calculate()];
+  }
+
+  // @todo we need to flesh this out
+  // For now it just checks that a license and copyright statements are present
+  calculate() {
+    const hasLicense = _.get(this.definition, 'licensed.license', false);
+    const hasCopyright = _.get(this.definition, 'licensed.copyright.statements[0]', false);
+    if (hasLicense && hasCopyright) {return 3;}
```

Comment:
formatting. Convention would have this be 
```javascript
   if (hasLicense && hasCopyright) 
     return 3;
```

Reply:
Can we use something like prettier? It eliminates formatting issues and let's people focus on the logic/business use of the code.

Reply:
James long gives a great prettier talk here (8 mins) https://www.youtube.com/watch?v=hkfBvpEfWdA

Reply:
totally fine with prettier or any other formatter as long as there is an agreed to format. Looking for consistency. I've been using VS Code and the default formatting (very simplistic) has been fine. My only two formatting criteria are
* no artificially short lines (80 chars is very 80s.
* minimize vertical space in general (e.g., don't put every property on its own line in object literals unless the line would be way too long)

I have a personal preference for no ';' but will not mind either way.

Will start a thread on the implementation channel...